src/web/components/ClientSideTokenGeneration/CstgAddDomainDialog.tsx

```diff
@@ -46,27 +45,16 @@ function CstgAddDomainDialog({
         message: 'The domains entered already exist.',
       });
     } else {
-      const invalidDomains: string[] = [];
-      let allValid = true;
-      uniqueDomains.forEach((newDomain) => {
-        if (!isValidDomain(newDomain)) {
-          invalidDomains.push(newDomain);
-          allValid = false;
-        }
-      });
-      if (!allValid) {
+      // filter for uniqueness (e.g. 2 different domains entered could have the same root-level domain)
+      const dedupedDomains = deduplicateStrings(uniqueDomains);
+      const invalidDomains = await onAddDomains(dedupedDomains, deleteExistingList);
+      if (invalidDomains.length > 0) {
         setError('root.serverError', {
```

Comment:
For my understanding, how does this `root.serverError` work? It's a bit confusing to me as it is not an error from the server 🤔

Reply:
This is the documentation for setError with React Hook Forms https://react-hook-form.com/docs/useform/seterror where the type is 'root.serverError' - these types of errors are good for resetting on each form submission

Reply:
thanks for linking that! what exactly do you mean by `root.serverError` being good for resetting on each form submission?

By reading the docs, I would have thought that we should be using the single error case, something like:
```
setError('newDomains', {
  type: 'manual',
  message: 'The domains entered already exist.',
});
```

The `root.serverError` examples seem to be genuine API calls that use the status code from the response from the backend. 

I do recognise that there is a slightly different UX in that the error appears underneath the input rather than at the top of the form, but I think that's fine: 
![image](https://github.com/IABTechLab/uid2-self-serve-portal/assets/137864392/b605c51d-6128-4339-b7d9-4ed9db3d7e07)

Am I missing something else?

Reply:
New ticket will be created to address this